account_ecotax/models/account_move_line.py

```diff
@@ -0,0 +1,126 @@
+# © 2014-2023 Akretion (http://www.akretion.com)
+#   @author Mourad EL HADJ MIMOUNE <mourad.elhadj.mimoune@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+
+
+class AcountMoveLine(models.Model):
+    _inherit = "account.move.line"
+
+    ecotax_line_ids = fields.One2many(
+        "account.move.line.ecotax",
+        "account_move_line_id",
+        string="Ecotax lines",
+        copy=True,
+    )
+    subtotal_ecotax = fields.Float(
+        digits="Ecotax", store=True, compute="_compute_ecotax"
+    )
+    ecotax_amount_unit = fields.Float(
+        digits="Ecotax",
+        string="Ecotax Unit.",
+        store=True,
+        compute="_compute_ecotax",
+    )
+
+    @api.depends(
+        "move_id.currency_id",
+        "ecotax_line_ids",
+        "ecotax_line_ids.amount_unit",
+        "ecotax_line_ids.amount_total",
+    )
+    def _compute_ecotax(self):
+        for line in self:
+            ecotax_ids = line.tax_ids.filtered(lambda tax: tax.is_ecotax)
+
+            if line.display_type == "tax" or not ecotax_ids:
+                continue
+            if line.display_type == "product" and line.move_id.is_invoice(True):
+                amount_currency = line.price_unit * (1 - line.discount / 100)
+                handle_price_include = True
+                quantity = line.quantity
+            else:
+                amount_currency = line.amount_currency
+                handle_price_include = False
+                quantity = 1
+            compute_all_currency = ecotax_ids.compute_all(
+                amount_currency,
+                currency=line.currency_id,
+                quantity=quantity,
+                product=line.product_id,
+                partner=line.move_id.partner_id or line.partner_id,
+                is_refund=line.is_refund,
+                handle_price_include=handle_price_include,
+                include_caba_tags=line.move_id.always_tax_exigible,
+            )
+            subtotal_ecotax = 0.0
+            for tax in compute_all_currency["taxes"]:
+                subtotal_ecotax += tax["amount"]
+
+            unit = quantity and subtotal_ecotax / quantity or subtotal_ecotax
+            line.ecotax_amount_unit = unit
+            line.subtotal_ecotax = subtotal_ecotax
+
```

Comment:
Why do we change `_compute_ecotax`?
In previous version of this module (`l10n_fr_ecotaxe`) we rely on `ecotaxe_line_ids.amount_unit` and `ecotaxe_line_ids.amount_total`. So, when `default_fixed_ecotax` is set on the classification it's applied.
```
unit = sum(line.ecotaxe_line_ids.mapped("amount_unit"))
subtotal_ecotaxe = sum(line.ecotaxe_line_ids.mapped("amount_total"))

line.update(
    {
        "ecotaxe_amount_unit": unit,
        "subtotal_ecotaxe": subtotal_ecotaxe,
    }
)
```
Some tests here are failing, because of this "lost" condition. Probably it should be refactored to take into account `default_fixed_ecotax`?

Reply:
I change `_compute_ecotax` because we use odoo taxe to compute ecotax. 
I'll fix test code.

>